ingot-macros/src/packet/mod.rs

```diff
@@ -862,17 +924,26 @@ impl StructParseDeriveCtx {
     pub fn gen_header_impls(&self) -> TokenStream {
         let ident = &self.ident;
         let validated_ident = self.validated_ident();
-        let base_bytes: usize = self
-            .chunk_layout
+        let base_size: ChunkSize = self.chunk_layout.iter().fold(
+            ChunkSize::default(),
+            |mut acc: ChunkSize, v: &ChunkState| {
+                if let ChunkState::FixedWidth { size, .. } = &v {
+                    acc.bytes += size.bytes;
+                    for (k, v) in size.zc_fields.iter() {
+                        *acc.zc_fields.entry(k.clone()).or_default() += *v;
+                    }
+                }
+                acc
+            },
+        );
+
+        let base_size_bytes = base_size.bytes;
+        let zc_field_sizes = base_size
+            .zc_fields
             .iter()
-            .map(|v| match v {
-                ChunkState::FixedWidth { size_bytes, .. } => *size_bytes,
-                ChunkState::VarWidth(_) => 0,
-                // NOTE: we should/can also include <ty>::MINIMUM_LENGTH here,
-                //       since that will stll be a constexpr.
-                ChunkState::Parsable(_) => 0,
```

Comment:
Since this note is being lost, could we add a `parsable_fields` to `ChunkSize` and use the same approach to include `quote!{ <#ty as ::ingot::types::HeaderLen>::MINIMUM_LENGTH * #n }` for each of those?

Reply:
Done in d8b159ff39e1.  A unit test would probably be good – are there any existing packet types that we could use?

Reply:
Thanks!

> A unit test would probably be good – are there any existing packet types that we could use?

Sadly not today, both places it comes up are `Repeated` (i.e., `MINIMUM_LENGTH` is `0`).